addons/sale_timesheet/models/sale_order_line.py

```diff
@@ -94,11 +92,10 @@ def _convert_qty_company_hours(self, dest_company):
         product_uom = self.product_uom_id
         if product_uom == self.env.ref('uom.product_uom_unit'):
             product_uom = self.env.ref('uom.product_uom_hour')
-        if product_uom.category_id == company_time_uom_id.category_id:
-            if product_uom != company_time_uom_id:
-                allocated_hours = product_uom._compute_quantity(self.product_uom_qty, company_time_uom_id)
-            else:
-                allocated_hours = self.product_uom_qty
+        if product_uom != company_time_uom_id:
```

Comment:
I would keep a check to see if the `product_uom` is related to the `company_time_uom_id`. Otherwise you could have allocated hours that are not even time-related.